crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -197,36 +197,20 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
 
     // TODO(joshie) TEMPORARY should be moved to trait providers
 
-    /// Traverse over changesets and plain state and recreate the [`PostState`]s for the given range
-    /// of blocks.
+    /// Unwind or peek at last N blocks of state
     ///
-    /// 1. Iterate over the [BlockBodyIndices][tables::BlockBodyIndices] table to get all
-    /// the transaction ids.
-    /// 2. Iterate over the [StorageChangeSet][tables::StorageChangeSet] table
-    /// and the [AccountChangeSet][tables::AccountChangeSet] tables in reverse order to reconstruct
-    /// the changesets.
-    ///     - In order to have both the old and new values in the changesets, we also access the
-    ///       plain state tables.
-    /// 3. While iterating over the changeset tables, if we encounter a new account or storage slot,
-    /// we:
-    ///     1. Take the old value from the changeset
-    ///     2. Take the new value from the plain state
-    ///     3. Save the old value to the local state
-    /// 4. While iterating over the changeset tables, if we encounter an account/storage slot we
-    /// have seen before we:
-    ///     1. Take the old value from the changeset
-    ///     2. Take the new value from the local state
-    ///     3. Set the local state to the value in the changeset
+    /// If UNWIND it set to true tip and latest state will be unwind
+    /// and returned back with all the blocks
     ///
-    /// If `TAKE` is `true`, the local state will be written to the plain state tables.
-    /// 5. Get all receipts from table
```

Comment:
these comments were really helpful, why did they get removed?

Reply:
Added it back.